File: 3_Longest_Substring_Without_Repeating_Characters/3_sliding_window_sol.c
#define MAX_ASCII_COUNT 128

int lengthOfLongestSubstring(char * s){
    int left = 0;
    int output = 0;
    int len = strlen(s);
    int seen[MAX_ASCII_COUNT];
    memset(seen, -1, sizeof(seen));

    for (int right = 0; right < len; right++) {
        if (seen[s[right]] != -1) {
            left = (left > seen[s[right]] + 1) ? left : seen[s[right]] + 1;
        }
        output = (output > right - left + 1) ? output : right - left + 1;
        seen[s[right]] = right;
    }
    return output;
}

File: Leetcode_problems/3_Longest_Substring_Without_Repeating_Characters/3_sliding_window_sol.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        seen = {}
        output = 0

        for right in range(len(s)):
            if s[right] in seen:
                left = max(left, seen[s[right]] + 1)
            output = max(output, right - left + 1)
            seen[s[right]] = right

        return output